### PR TITLE
Node week 1

### PR DESCRIPTION
**Pull request by elicarcar**
I couldn't pass the test, I don't know what I'm doing wrong

### CODE REVIEW

**Comment by remarcmij** [comment]
Hi @elicarcar, did you install the required dependencies as listed here?

https://github.com/HackYourFuture/Node.js/blob/master/week1/homework/README.md#installing-dependencies

Your PR tests fine on my computer:

```
PS C:\Users\jimcr\dev\hackyourfuture\Node.js\week1\homework> npm test

> hyf-node-week-1-homework@2.0.0 test C:\Users\jimcr\dev\hackyourfuture\Node.js\week1\homework
> ava --verbose

  √ /reset returns 10
  √ /state returns 10
  √ /add returns 11
  √ /subtract returns 9
  √ querying undefined URL returns 404 Not Found
  √ /add, /reset returns 10
  √ /subtract, /reset returns 10
  √ /add, /add, /state, /add, /subtract returns 12
  √ /subtract, /subtract, /reset, /add, /state, /subtract, /add returns 11
  √ /add, /add, /add, /add, /add, /add, /add, /add, /add, /add, /state returns 20
  √ /subtract, /subtract, /subtract, /subtract, /subtract, /subtract, /subtract, /subtract, /subtract, /subtract, /state returns 0

  11 tests passed
```

File: week1/homework/src/server.js [hunk 1]
```diff
@@ -6,10 +6,33 @@ const http = require('http');
...
-    // TODO: Write your homework code here
+    function responseWrite(statusCode, result = {}) {
+      response.writeHead(statusCode, { 'Content-Type': 'application/json' });
+      response.write(JSON.stringify(result));
+    }
+
+    switch (request.url) {
+      default:
+        responseWrite(404, { error: 'Not found' });
+        break;
```
Comment (Meno-101) [inline week1/homework/src/server.js]: for a best practice and more legibility I would go for keeping this `default` case at the end of my code.

File: week1/homework/src/server.js [hunk 2]
```diff
@@ -6,10 +6,33 @@ const http = require('http');
  * will fail.
  */
 function createServer(port) {
+  // TODO: Write your homework code here
   let state = 10;
-
   const server = http.createServer((request, response) => {
-    // TODO: Write your homework code here
+    function responseWrite(statusCode, result = {}) {
```
Comment (Meno-101) [inline week1/homework/src/server.js]: Nice DRY one!

File: week1/homework/src/server.js [hunk 3]
```diff
@@ -6,10 +6,33 @@ const http = require('http');
...
+    switch (request.url) {
+      default:
+        responseWrite(404, { error: 'Not found' });
+        break;
+      case '/state':
+        responseWrite(200, { state });
+        break;
+
+      case '/add':
+        responseWrite(200, { state: (state += 1) });
```
Comment (Meno-101) [inline week1/homework/src/server.js]: It's all about readability, this way of reassigning your variable will work just fine, but for a personal taste leaving this `state += 1` on it's own line will make it improvement in that point.
```javascript
state += 1
responseWrite(200, { state });
```

File: week1/homework/src/server.js (hunk 1 above)
Comment (elicarcar) [reply week1/homework/src/server.js]: owww ok I will change it right away,

File: week1/homework/src/server.js (hunk 3 above)
Comment (elicarcar) [reply week1/homework/src/server.js]: thank you, then I will change that as well

**Pull request closed without merging**